### PR TITLE
Make pytket-qir a required dependency of pytket-quantinuum.

### PR DESCRIPTION
**Pull request by cqc-alec**
# Description

I forget why it wasn't already, but I don't see why it can't be now.

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 1]
```diff
@@ -883,34 +878,17 @@ def process_circuits(
                     for name, count in Counter(bit.reg_name for bit in c0.bits).items():
                         for i in range(count):
                             results_selection.append((name, i))
-                    try:
```
Comment (cqc-melf) [inline pytket/extensions/quantinuum/backends/quantinuum.py:886]: Should we still do a try and catch for possible errors and give more specific details that the conversion failed and they could try the qasm way if they want to?

**Review by cqc-melf — COMMENTED** [review]
Looks good, do you want to add something to the changelog?

I think the reason that we have not added this was that we wanted to think if we want to move this over into pytket and have it there next to the qasm conversion? Has this decision been made?

**Comment by cqc-alec** [comment]
> Looks good, do you want to add something to the changelog?

Yes, I should add something to the changelog.

> I think the reason that we have not added this was that we wanted to think if we want to move this over into pytket and have it there next to the qasm conversion? Has this decision been made?

Not really, but as it's not needed anywhere else (yet) I think it's fine to have the dependency here. If and when we decide we want QIR conversion in pytket it will be easy enough to add the dependency there and remove the explicit requirement from here.

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 1 above)
Comment (cqc-alec) [reply pytket/extensions/quantinuum/backends/quantinuum.py:886]: Currently the only message we print on error is "try installing pytket-qir", and that shouldn't be necessary now. Better error messages could be added in another PR, though the error raised by `pytket_to_qir()` should contain at least some useful information.

File: docs/changelog.rst [hunk 2]
```diff
@@ -4,6 +4,11 @@ Changelog
 0.29.0 (January 2024)
```
Comment (cqc-melf) [inline docs/changelog.rst:4]:
```suggestion
0.30.0 (January 2024)

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 1 above)
Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/quantinuum.py:886]: Yes, but that error will not contain any hints pytket-quantinuum specific. Happy to do this in a separate PR.

**Review by cqc-melf — DISMISSED** [review]

File: docs/changelog.rst (hunk 2 above)
Comment (cqc-melf) [reply docs/changelog.rst:4]: Not sure if this is required, we will clean this up before doing a release anyway. Feel free to merge without this.

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**